src/EVM/Expr.hs

```diff
@@ -628,6 +633,63 @@ getAddr (GVar _) = error "cannot determine addr of a GVar"
 
 -- ** Whole Expression Simplification ** -----------------------------------------------------------
 
+-- Calls prop simplification for the appropriate types of Expr
+applyToProps :: ([Prop] -> [Prop]) -> Expr b -> Expr b
+applyToProps f expr = case expr of
+  Failure a b c -> Failure (f a) b c
+  Partial a b c -> Partial (f a) b c
+  Success a b c d -> Success (f a) b c d
+  ITE (Lit 0) _ c -> applyToProps f c
+  ITE (Lit _) b _ -> applyToProps f b
+  ITE a b c-> ITE a (applyToProps f b) (applyToProps f c)
+  a -> a
+
+remRedundantProps :: [Prop] -> [Prop]
+remRedundantProps p = collapseFalse . filter (\x -> x /= PBool True) . nubOrd $ p
+  where
+    collapseFalse :: [Prop] -> [Prop]
+    collapseFalse ps = if isJust $ find (== PBool False) ps then [PBool False] else ps
+
+
+simplifyProp :: Prop -> Prop
```

Comment:
whats the difference between this and `evalProp`?

Reply:
Oh!! NICE catch! Actually, this does less AND more at the same time :D OK, let me merge the two and use evalProp here :)

Reply:
NICE! So much better, thank you! I am sometimes super-blind. You are so meticulous, it's so much better this way.

So b44c31a3d43f36046949d0c7f012bebad932e27e remedies this -- now using `evalProp`, and also improved `evalProp`. I have something to learn from the way `evalProp` was written, some constant simplifications were so much better done!

Reply:
I'm resolving because this has been fixed.